pallets/file-system/src/lib.rs

```diff
@@ -466,13 +464,20 @@ pub mod pallet {
             origin: OriginFor<T>,
             location: FileLocation<T>,
             root: FileKey<T>,
-            proof: Proof<T>,
+            forest_proof: ForestProof<T>,
+            key_proof: KeyProof<T>,
```

Comment:
The reason why I purposely separated this was because the "merged" `Proof<T>` type in the proofs dealer pallet, is used for a WHOLE proof. Basically, the key proofs in a `Proof<T>` correspond to the keys proven in the forest proof of the `Proof<T>`. In this case, the `forest_proof` will have a proof of non-inclusion of the file key being confirmed, i.e. the two neighbouring keys. `key_proof`, on the other hand, will have a proof of the file being added. Basically they don't correspond to each other, and that's why I have them separate.